From 64e6bd333d221709c371de48604eaf06a982e873 Mon Sep 17 00:00:00 2001
From: Shawn Willden <swillden@google.com>
Date: Thu, 26 Mar 2020 14:40:59 -0600
Subject: [PATCH 05/15] Remove beginOp and IOperation

The way I planned for this to work doesn't work.  We'll revisit in
Keymaster5.  For now, removing IOperation and beginOp.

Bug: 152536287
Test: VtsHalKeymasterV4_1TargetTest
Change-Id: Iea731d3986c933a1ea43603c60af21aa57ae8bcb
---
 ng/include/AndroidKeymaster41Device.h | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/ng/include/AndroidKeymaster41Device.h b/ng/include/AndroidKeymaster41Device.h
index 8e6d0f8..68bbcf6 100644
--- a/ng/include/AndroidKeymaster41Device.h
+++ b/ng/include/AndroidKeymaster41Device.h
@@ -21,7 +21,6 @@
 #include <android/hardware/keymaster/4.1/IKeymasterDevice.h>
 #include <android/hardware/keymaster/4.1/types.h>
 #include <hidl/Status.h>
-#include <keymasterV4_1/Operation.h>
 
 #include "AndroidKeymaster4Device.h"
 
@@ -47,7 +46,6 @@ using ::android::hardware::keymaster::V4_0::OperationHandle;
 using ::android::hardware::keymaster::V4_0::SecurityLevel;
 using ::android::hardware::keymaster::V4_0::VerificationToken;
 using ::android::hardware::keymaster::V4_1::IKeymasterDevice;
-using ::android::hardware::keymaster::V4_1::IOperation;
 using ::android::hardware::keymaster::V4_1::Tag;
 
 using V41ErrorCode = ::android::hardware::keymaster::V4_1::ErrorCode;
@@ -71,17 +69,6 @@ class AndroidKeymaster41Device : public IKeymasterDevice, public V4_0::ng::Andro
                                       const VerificationToken& /* verificationToken */) override;
     Return<V41ErrorCode> earlyBootEnded() override;
 
-    Return<void> beginOp(KeyPurpose purpose, const hidl_vec<uint8_t>& key,
-                         const hidl_vec<KeyParameter>& inParams, const HardwareAuthToken& authToken,
-                         beginOp_cb _hidl_cb) override {
-        return super::begin(
-            purpose, key, inParams, authToken,
-            [&](auto hidl_err, auto hidl_params, auto hidl_handle) {
-                _hidl_cb(convert(hidl_err), hidl_params,
-                         new ::android::hardware::keymaster::V4_1::support::Operation(hidl_handle));
-            });
-    }
-
     Return<void> getHardwareInfo(super::getHardwareInfo_cb _hidl_cb) override {
         return super::getHardwareInfo(_hidl_cb);
     }
-- 
2.17.1

